core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -584,16 +605,27 @@ public void runInContext() {
 
     @Override
     public void closed(final Status status) {
+      // For cancellations, promptly inform any users of the context that their work should be
+      // aborted. Otherwise, we can wait until pending work is done.
+      if (!status.isOk()) {
+        // The callExecutor might be busy doing user work. To avoid waiting, use the alternate
+        // cancelExecutor.
```

Comment:
To be more accurate, you may want to say "The callExecutor is serializing executor. In order to notify tasks that are already in the callExecutor, we use an executor that is not serializing".